### PR TITLE
cody-gateway, sourcegraph: disable code completions for OpenAI

### PR DESCRIPTION
**Pull request by bobheadxi**
Using OpenAI for Cody code completions is currently [untested and unsupported](https://sourcegraph.slack.com/archives/CN4FC7XT4/p1690986792111129?thread_ts=1690931810.610299&cid=CN4FC7XT4). Despite this, we have [at least one customer generating very large numbers of OpenAI code completions requests](https://sourcegraph.slack.com/archives/C03JR7S7KRP/p1691438665847449?thread_ts=1689373762.522179&cid=C03JR7S7KRP) - enough to get us [rate limited by OpenAI](https://sourcegraph.sentry.io/issues/4372449189/?alert_rule_id=14416305&alert_type=issue&project=4505201451270144).

This change:

- Makes Cody Gateway reject OpenAI code completions requests (responding with 501 not implemented)
- Prevents Sourcegraph from generating OpenAI code completion requests

## Test plan

CI

### CODE REVIEW

File: internal/completions/client/azureopenai/openai.go [hunk 1]
```diff
@@ -34,6 +34,10 @@ func (c *azureCompletionClient) Complete(
 	feature types.CompletionsFeature,
 	requestParams types.CompletionRequestParameters,
 ) (*types.CompletionResponse, error) {
+	if feature == types.CompletionsFeatureCode {
+		return nil, errors.Newf("%q for OpenAI is currently not supported")
```
Comment (eseliger) [inline internal/completions/client/azureopenai/openai.go:38]:
```suggestion
		return nil, errors.Newf("%q for Azure OpenAI is currently not supported")
```

**Review by eseliger — APPROVED** [review]
Works for me, but I'll defer to other people tagged for review here to make the call if this is standing in the way of some other efforts.

**Comment by bobheadxi** [comment]
[Beyang indicated disabling is likely the right way to go](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1691445236009229?thread_ts=1691443942.596769&cid=C04MSD3DP5L), but will wait for another approval here 👍

**Review by philipp-spiess — APPROVED** [review]
I think this makes sense, yeah. 

Should we add to the docs that enabling OpenAI disables code completions completely? Just so that people aren't confused.

**Pull request merged**